packages/react-aria-components/src/utils.tsx

```diff
@@ -151,7 +151,7 @@ export function useContextProps<T, U, E extends Element>(props: T & SlotProps, r
   }
   // @ts-ignore - TS says "Type 'unique symbol' cannot be used as an index type." but not sure why.
   let {ref: contextRef, [slotCallbackSymbol]: callback, ...contextProps} = ctx;
-  let mergedRef = useObjectRef(mergeRefs(ref, contextRef));
+  let mergedRef = useObjectRef(useMemo(() => mergeRefs(ref, contextRef), [ref, contextRef]));
```

Comment:
if merge refs should always be memo'd, it looks like there are a few other instances in our react-aria-components as well as datepicker and link where we do not do this